### PR TITLE
Add RSS feeds for categories and tags

### PR DESCRIPTION
**Pull request by ksdhans**
This patch adds RSS feeds for tags and categories (for issue #382). The RSS feeds are at the following URLs:
- */category/<category-name>/rss
- */tag/<tag-name>/rss

### CODE REVIEW

File: code/model/Blog.php [hunk 1]
```diff
@@ -1093,4 +1098,30 @@ public function getRSSLink()
...
     }
+    
+    /**
+     * Displays an RSS feed of the given blog posts.
+     *
+     * @return string
+     */
+    public function rssFeed($blogPosts)
+    {
+        $rss = new RSSFeed($blogPosts, $this->Link() . 'category/general/', $this->MetaTitle, $this->MetaDescription);
```
Comment (micmania1) [inline code/model/Blog.php]: This still has hard-coded url segments.

File: code/model/Blog.php [hunk 2]
```diff
@@ -1093,4 +1098,30 @@ public function getRSSLink()
...
+
+        $this->extend('updateRss', $rss);
+
+        return $rss->outputToBrowser();
+    }
+    
+    /** 
+     * Returns true if the $Rss sub-action for categories/tags has been set to "rss"
+     */
+     private function isRSS() {
```
Comment (micmania1) [inline code/model/Blog.php]: Could you drop the bracket to the next line please (psr-2) and document the return type?

File: code/model/Blog.php [hunk 3]
```diff
@@ -1093,4 +1098,30 @@ public function getRSSLink()
...
+     	 $rss = $this->request->param('Rss');
+     	 if($rss && strcasecmp($rss, "rss") == 0) {
```
Comment (micmania1) [inline code/model/Blog.php]: Could you change `if ($rss ...` to `if(is_string($rss)...`. strcasesmp *can* throw an error in some circumstances - this is just a little extra safety.

File: code/model/Blog.php [hunk 4]
```diff
@@ -1093,4 +1098,30 @@ public function getRSSLink()
     {
         return $this->Link('rss');
     }
+    
+    /**
+     * Displays an RSS feed of the given blog posts.
+     *
+     * @return string
+     */
+    public function rssFeed($blogPosts)
```
Comment (micmania1) [inline code/model/Blog.php]: Could you make this protected please?

**Comment by micmania1** [comment]
Thanks for the contribution @ksdhans.

I've left a few small comments but other than that i'd be happy with this being merged.

In future we'd probably like to split out tags and categories to their own controller with their own `rss` method as to avoid nasty method names and also the `isRss` check which feels a little hacky. (we could have `isJson`, `isXml` etc.) which would eventuate in quite nasty if statements.

Its good to see progress and marking `isRssFeed` as private is a good idea - this should allow us to make the above changes without any BC issues in future 👍 .

I'll wait for any other feedback and your updates before merging.

**Comment by ksdhans** [comment]
Okay, I've made the changes. However, I've just discovered that the RSS feed's link/atom:link always points to the root URL. So it's not quite right yet.

**Comment by ksdhans** [comment]
The link URLs are now correct for category/tag rss feeds, so I think that it's all working as it should.

File: code/model/Blog.php [hunk 5]
```diff
@@ -1093,4 +1098,34 @@ public function getRSSLink()
...
+
+        return $rss->outputToBrowser();
+    }
+    
+    /** 
+     * Returns true if the $Rss sub-action for categories/tags has been set to "rss"
+     */
+     private function isRSS() 
+     {
+     	 $rss = $this->request->param('Rss');
```
Comment (micmania1) [inline code/model/Blog.php]: The formatting seems a bit strange here (tabs/spaces) - not really sure what's going on.

**Comment by micmania1** [comment]
@ksdhans just one small thing.

@silverstripe/module-maintainers-blog-comments-and-related-modules anybody got any input before merging?

File: code/model/Blog.php (hunk 5 above)
Comment (ksdhans) [inline code/model/Blog.php]: Yeah, that's the code editor that I'm using being weird (and me not noticing). Sorry about that.

**Comment by micmania1** [comment]
@ksdhans merged! Thanks 👍

**Pull request merged**